### PR TITLE
newsear

### PR DESCRIPTION
**Pull request by andyngdz**

### CODE REVIEW

File: challenge-react/src/App.js [hunk 1]
```diff
@@ -1,111 +1,111 @@
...
+    if (infoMessage) {
+      const timer = setTimeout(() => {
+        setInfoMessage(undefined);
+      }, 5000);
+      return () => clearTimeout(timer);
+    }
+  }, [infoMessage]);
+
+  useEffect(() => {
+    fetchCharities();
```
Comment (andyngdz) [inline challenge-react/src/App.js:57]: Screen Shot 2021-01-25 at 10 33 09

- Where are the dependencies of these functions?

File: challenge-react/src/App.js [hunk 2]
```diff
@@ -1,111 +1,111 @@
...
+      .then((resp) => resp.json())
+      .then((charities) => setCharities(charities));
+  };
+
+  const fetchPayments = () => {
+    fetch(`${API_URL}/payments`)
+      .then((resp) => resp.json())
+      .then((payments) => {
+        const totalDonation = payments.reduce(
+          (totalDonation, payment) => (totalDonation += payment.amount),
```
Comment (andyngdz) [inline challenge-react/src/App.js:72]: Screen Shot 2021-01-25 at 10 36 00

It seems you forgot to install eslint

File: challenge-react/src/components/card.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+  title,
+}) => {
+  const [image, setImage] = useState();
+  const [payAmount, setPayAmount] = useState(DEFAULT_PAY_AMOUNT);
+
+  useEffect(() => {
+    setPayAmount(DEFAULT_PAY_AMOUNT);
+  }, [displayOverlay]);
+
+  useEffect(() => {
```
Comment (andyngdz) [inline challenge-react/src/components/card.js:106]: Not a good practice to import the image here.
You can create a file: image.js

```
import Image1 from '../assets/images/baan-kru-noi.jpg`
import Image2 ....

export { Image1, Image2 }
```

Then from here, you can import image.js and show it

React will bundle your image  and will load faster

**Comment by andyngdz** [comment]
According to the requirements. These points are missing:
- Refactor the code to be more readable and enhance reusability.

Bonus:
- Write helpers or components unit tests with jest. -> Important

In the other hand, feel free to:
- Improve the design to have better UI and UX.
- Re-organize the codebase.
- Update webpack config. (It lacks `plugin-transform-runtime` check my PR to fix this: https://github.com/opn-ooo/challenges/pull/13/commits/3dbd08c64c5b17c672287e785912be985d1511ab)
- Take reasonable time to complete the challenge, no need to rush. -> No need to rush, it's missing the most important points.

File: challenge-react/src/components/card.js [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+  id,
+  imageName,
+  onSubmit,
+  setDisplayOverlay,
+  title,
+}) => {
+  const [image, setImage] = useState();
+  const [payAmount, setPayAmount] = useState(DEFAULT_PAY_AMOUNT);
+
+  useEffect(() => {
```
Comment (EpicHigh) [inline challenge-react/src/components/card.js:102]: Separate logic: why did you set the amount on the handle submit and useEffect twice.

File: challenge-react/src/components/card.js [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+    setPayAmount(DEFAULT_PAY_AMOUNT);
+  }, [displayOverlay]);
+
+  useEffect(() => {
+    import(`../assets/images/${imageName}`).then((image) => {
+      setImage(image.default);
+    });
+  }, []);
+
+  const handleSubmit = () => {
```
Comment (EpicHigh) [inline challenge-react/src/components/card.js:112]: Separate logic: why did you set the amount on the handle submit and useEffect twice.

File: challenge-react/src/components/card.js [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+    onSubmit(payAmount);
+    setDisplayOverlay(undefined);
```
Comment (EpicHigh) [inline challenge-react/src/components/card.js:114]: `setDisplayOverlay(DEFAULT_PAY_AMOUNT)`

**Review by EpicHigh — COMMENTED** [review]
No test cases.

**Pull request closed without merging**